src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -18,102 +18,99 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Runtime.CompilerServices;
+using Nethermind.Core.Collections;
 
 namespace Nethermind.TxPool.Collections
 {
-    public class SortedPool<TKey, TValue>
+    public class SortedPool<TKey, TValue, TGroup>
     {
         private readonly int _capacity;
-        protected readonly Comparison<TValue> Comparison;
-        protected readonly Dictionary<TKey, LinkedListNode<KeyValuePair<TKey, TValue>>> CacheMap;
-        protected readonly LinkedList<KeyValuePair<TKey, TValue>> LruList;
-
-        public SortedPool(int capacity, Comparison<TValue> comparison)
+        private readonly IComparer<TValue> _comparerWithIdentity;
+        private readonly Func<TValue, TGroup> _groupMapping;
+        private readonly IDictionary<TGroup, ICollection<TValue>> _buckets;
+        private readonly DictionarySortedSet<TValue, TKey> _sortedValues;
+        private readonly IDictionary<TKey, TValue> _cacheMap;
+        
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="capacity">Max capacity</param>
+        /// <param name="comparerWithIdentity">Comparer to sort items. It must differentiate items by their identity or some items will be lost.</param>
```

Comment:
this sounds like it is breaking the encapsulation, you define the interface and then sdd some comments to tell how it should behave

Reply:
This is more a hint to user about expected behaviour of this class. The user of this class decides what is identity. This only hints that it will use this comparer as identity comparer, which is often not obvious when writing IComparer when you only focus on sorting and not on identity.